client/src/app/appCtrl.js

```diff
@@ -57,16 +59,39 @@ angular.module('app').controller('appCtrl', ['$scope', '$http', '$timeout', 'app
             slidingWindow = true;
             $scope.view.windowing.show = true;
           }
-          streamRemoteFile($scope.view.filePath);
+          downloadFile($scope.view.filePath, "streamRemote");
         });
       } else {
-        downloadFile($scope.view.filePath);
+        downloadFile($scope.view.filePath, "download");
       }
-    }, function() {
-      downloadFile($scope.view.filePath);
-    });
+     }, 
+     function() {
+      downloadFile($scope.view.filePath, "download");
+     }
+    );
   };
 
+  // handle downloading (or Browsing) a local file
+  $scope.getLocalFile = function(event) {
+    $scope.view.filePath = event.target.files[0].name;
+    if (event.target.files[0].size > appConfig.MAX_FILE_SIZE) {
+      slidingWindow = true;
+      $scope.view.windowing.show = true;
+    }
+    $scope.view.loading = true;
+    downloadFile(event.target.files[0], "streamLocal");
+  };
+
+  // main "load" function that supports both URL/local file
+  $scope.loadFile = function(event) {
```

Comment:
this can be removed, but I thought if you could change the buttons to work with only this method? So there'd be only 1 button "Load". Do you think it'd work?